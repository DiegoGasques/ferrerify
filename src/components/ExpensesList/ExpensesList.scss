.ExpensesList {
}

.ExpensesList__header {
  background: #303030;
  padding: 0.7rem 1.5rem;
  margin: 1.5rem 0;
  -webkit-box-shadow: 0px 4px 15px -5px rgba(0, 0, 0, 0.5),
    inset 0px 0px 15px -9px rgba(0, 0, 0, 0.6);
  box-shadow: 0px 4px 15px -5px rgba(0, 0, 0, 0.5),
    inset 0px 0px 15px -9px rgba(0, 0, 0, 0.6);
}

.ExpensesList__header h2 {
  color: white;
  font-size: 2rem;
}

.ExpensesList__content {
  margin: 1.5rem 0;
}
