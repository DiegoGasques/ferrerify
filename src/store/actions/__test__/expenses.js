import { addExpense, removeExpense, editExpense } from "../expenses.actions.js";

it("Test for addExpense action creators correct output", () => {
  const newExpense = {
    description: "New expense",
    note: "This is a big expense",
    createdAt: 1000000,
    amount: 15000
  };
  const action = addExpense(newExpense);

  expect(action).toEqual({
    type: "ADD_EXPENSE",
    expense: {
      ...newExpense,
      id: expect.any(String)
    }
  });
});

it("Test for removeExpense action creators correct output", () => {
  const id = "123";
  const action = removeExpense(id);
  expect(action).toEqual({
    type: "REMOVE_EXPENSE",
    id
  });
});

it("Test for editExpense action creators correct output", () => {
  const id = "123";
  const updates = { note: "The updated note" };
  const action = editExpense(id, updates);
  expect(action).toEqual({
    type: "EDIT_EXPENSE",
    id,
    updates
  });
});
