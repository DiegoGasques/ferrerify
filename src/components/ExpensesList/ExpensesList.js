import React from "react";
import ExpenseListItem from "./ExpensesListItem/ExpensesListItem";
import { connect } from "react-redux";
import { sortByMatchSelector } from "../../store/selectors/expenses.selector";
import Filters from "../Filters/Filters";

export const ExpensesList = ({ expenses }) => {
  const generate = () =>
    expenses.map(expense => <ExpenseListItem key={expense.id} {...expense} />);

  return (
    <div className="ExpensesList">
      <div className="ExpensesList__header">
        <h2>Expenses</h2>
      </div>
      <Filters />
      <div className="ExpensesList__content">
        {!!expenses ? generate() : <div>Loading...</div>}
      </div>
    </div>
  );
};

const mapStateToProps = state => ({
  expenses: sortByMatchSelector(state)
});

export default connect(
  mapStateToProps,
  null
)(ExpensesList);
