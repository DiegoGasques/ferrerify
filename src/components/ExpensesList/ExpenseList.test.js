import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { ExpensesList } from "./ExpensesList";
import { expensesArr } from "../../fixtures/expenses";

describe("[COMPONENT] ExpensesList", () => {
  it("Tests for changes in the rendered component with expenses", () => {
    const wrapper = shallow(<ExpensesList expenses={expensesArr} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("Tests for changes in the rendered component with no expense", () => {
    const wrapper = shallow(<ExpensesList expenses={[]} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
