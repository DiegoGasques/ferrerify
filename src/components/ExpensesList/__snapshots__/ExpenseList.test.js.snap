// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[COMPONENT] ExpensesList Tests for changes in the rendered component with expenses 1`] = `
<div
  className="ExpensesList"
>
  <div
    className="ExpensesList__header"
  >
    <h2>
      Expenses
    </h2>
  </div>
  <Connect(Filters) />
  <div
    className="ExpensesList__content"
  >
    <Connect(ExpensesListItem)
      amount={150000}
      createdAt={0}
      description="Rent"
      id="1"
      key="1"
      note="Monthly payment"
    />
    <Connect(ExpensesListItem)
      amount={1000}
      createdAt={-345600000}
      description="Car washing"
      id="2"
      key="2"
      note="Non planned expense"
    />
    <Connect(ExpensesListItem)
      amount={22000}
      createdAt={345600000}
      description="Groceries"
      id="3"
      key="3"
      note="Weekly expense"
    />
  </div>
</div>
`;

exports[`[COMPONENT] ExpensesList Tests for changes in the rendered component with no expense 1`] = `
<div
  className="ExpensesList"
>
  <div
    className="ExpensesList__header"
  >
    <h2>
      Expenses
    </h2>
  </div>
  <Connect(Filters) />
  <div
    className="ExpensesList__content"
  />
</div>
`;
